pyfarm/jobtypes/core/jobtype.py

```diff
@@ -932,6 +932,13 @@ def set_task_state(self, task, state, error=None, dissociate_agent=False):
             logger.error("Not setting task to failed: agent is shutting down.")
 
         else:
+            assignment_id = self.assignment["id"]
+            assignment = config["current_assignments"].get(assignment_id, None)
+            if assignment:
+                task_ = next(x for x in assignment["tasks"] if
```

Comment:
Hmm, I think a straight for loop with a break would be just as efficient but more readable.  `x` should also be named something more human friendly too.